### ISSUE TITLE
The PoC produces random result due to the ghosted file being created asynchronous

### ISSUE THREAD

**Issue by ladislav-zezula** [hasherezade/process_ghosting#3]
At [main.cpp(27)](https://github.com/hasherezade/process_ghosting/blob/043e00f4365d0400eb42b01c25ed8a06c7ea004f/main.cpp#L27), you are calling `NtOpenFile` without `SYNCHRONIZE` and `FILE_SYNCHRONOUS_IO_NONALERT` flags. This means that the file is open/created for asynchronous access. Any read/write operation is pended, and may be completed asynchronously if the operating system decides to do so. Usually, it happens after reboot, when the file is not in the system cache.

Then, at [main.cpp(59)](https://github.com/hasherezade/process_ghosting/blob/043e00f4365d0400eb42b01c25ed8a06c7ea004f/main.cpp#L59), you are calling NtWriteFile without waiting for the result.

TLDR: The Proof-of-concept will produce random results.

**Comment by hasherezade** [comment]
thank you for the heads-up, fixed
